pkg/kubelet/kubelet_test.go

```diff
@@ -4651,3 +4652,177 @@ func TestGenerateAPIPodStatusWithReasonCache(t *testing.T) {
 		verifyStatus(t, i, apiStatus, test.expectedState, test.expectedLastTerminationState)
 	}
 }
+
+// testPodAdmitObserver is a lifecycle.PodAdmitObserver that is used for testing.
+type testPodAdmitObserver struct {
+	// list of pods to reject.
+	podsToReject []*api.Pod
+}
+
+// Admit rejects all pods in the podsToReject list with a matching UID.
+func (a *testPodAdmitObserver) Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult {
+	for _, podToReject := range a.podsToReject {
+		if podToReject.UID == attrs.Pod.UID {
+			return lifecycle.PodAdmitResult{Admit: false, Reason: "Rejected", Message: "Pod is rejected"}
+		}
+	}
+	return lifecycle.PodAdmitResult{Admit: true}
+}
+
+// Test verifies that the kubelet invokes an admission handler during HandlePodAdditions.
+func TestHandlePodAdditionsInvokesPodAdmitObservers(t *testing.T) {
```

Comment:
I think we can retire a bunch of individual pod admission tests in kubelet_test.go after this lands, since we share the predicate library with the scheduler :)